src/utilities/modules/dispensary/refund.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import { createRecord } from '../../../database/utilities';
+import { UIDatabase } from '../../../database';
+
+/**
+ * Utility to refund a collection of TransactionBatch records. Creates a
+ * single CustomerCredit and for each TransactionBatch that should be
+ * refund, a RefundLine record, which enters the batch back into stock
+ * and adds credit for the patient.
+ *
+ *
+ * @param {User}               currentUser The currently logged in user.
+ * @param {Name}               patient     The patient to refund to
+ * @param {TransactionBatch[]} batches     Array of batches to be refund
+ */
+export const refund = (currentUser, patient, batches) => {
+  const forSamePatient = batches.every(({ transaction }) => {
+    const { otherPartyName } = transaction;
+    return patient.id === otherPartyName.id;
+  });
+
+  if (!batches.length) throw new Error('Trying to refund void');
+  if (!forSamePatient) throw new Error('Batches for different patients!');
+
+  const sumOfCredit = batches.reduce((acc, { total }) => acc + total, 0);
```

Comment:
are the refunded transactions marked in some way, so they can't be refunded again?

Reply:
No. Need to make another desktop issue to find out different/correct behaviour. A new field may need to be added to `[trans_line]` or something as you can just refund as many times as you like.

Can even refund the same item multiple times in the same credit!

![image](https://user-images.githubusercontent.com/35858975/69770743-0b243680-11ef-11ea-9045-e7a2faedcdf0.png)

Reply:
Hope you don't mind me addressing this elsewhere, possibly after a discussion!

Reply:
yes, I think it needs to be sorted separately - and especially because desktop needs a tidy up first!